velox/functions/prestosql/DateTimeImpl.h

```diff
@@ -42,13 +42,16 @@ FOLLY_ALWAYS_INLINE std::optional<Timestamp> fromUnixtime(double unixtime) {
 
   static const int64_t kMax = std::numeric_limits<int64_t>::max();
   static const int64_t kMin = std::numeric_limits<int64_t>::min();
+  static const uint64_t kSignedMax = (1ull << 63);
```

Comment:
static const uint64_t kSignedMax = kMax;

Reply:
kSignedMax is kMax +1  guess.

Reply:
can you add a comment explaining that and the issue so people do not revert 
also, kSignedMax naming got me lost, maybe wen can remove the variable and do the following in the check 
```
// converting kMax to double result into issues related to ... but... etc
  if (UNLIKELY((static_cast<unint64_t>(kMax)+1) - unixtime <= 1)) {
```

Reply:
@laithsakka can you take a look now?